common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -316,6 +383,19 @@ static ByteBuffer allocateDirectNoCleaner(int capacity) {
         return newDirectBuffer(UNSAFE.allocateMemory(capacity), capacity);
     }
 
+    static byte[] allocateUninitializedArray(int size) {
+        if (ALLOCATE_ARRAY_METHOD == null) {
```

Comment:
typically `PlatformDependent` takes care of calling methods in `PlatformDependent0` only if appropriate, and `PlatformDependent0` fails if it is called in an incorrectly initialized environment. Can we follow that convention here too? That will avoid duplicate logic checks in `PlatformDependent` and here.